File: backend/domain/workout.py
from enum import Enum
from typing import Optional

from sqlmodel import Field, SQLModel


class DifficultyLevel(str, Enum):
    beginner = "beginner"
    intermediate = "intermediate"
    advanced = "advanced"


class Workout(SQLModel, table=True):
  id: Optional[int] = Field(default=None, primary_key=True)
  
  title: str
  description: str
  main_goal: str                          
  workout_type: str                       
  training_level: DifficultyLevel         
  program_duration_weeks: Optional[int] = Field(default=None)  
  days_per_week: Optional[int] = Field(default=None)           
  time_per_workout: Optional[str] = Field(default=None)        
  equipment_required: Optional[str] = Field(default=None)      
  target_gender: Optional[str] = Field(default=None)           

File: front/src/domain/workout.ts
export type Modality = 'bodyweight' | 'equipment'

export interface Exercise {
  name: string
  sets: number
  reps: string
  rest_seconds?: number
  muscle_group?: string
  notes?: string
}

export interface WorkoutDay {
  day: number
  focus: string
  exercises: Exercise[]
}

export interface WorkoutPlan {
  id?: number
  title: string
  description?: string
  main_goal: string
  workout_type: string
  training_level: string
  program_duration_weeks?: number | null
  days_per_week?: number | null
  time_per_workout?: string | null
  equipment_required?: string | null
  target_gender?: string | null
  days: WorkoutDay[]
}
